### PR TITLE
[Path] export a camotics tooltable

### PR DESCRIPTION
**Pull request by sliptonic**
https://github.com/CauldronDevelopmentLLC/CAMotics/issues/335#issuecomment-1180062475

- [x]  Your pull request is confined strictly to a single module. That is, all the files changed by your pull request are either in `App`, `Base`, `Gui` or one of the `Mod` subfolders. If you need to make changes in several locations, make several pull requests and wait for the first one to be merged before submitting the next ones
- [x]  In case your pull request does more than just fixing small bugs, make sure you discussed your ideas with other developers on the FreeCAD forum
- [x]  Your branch is [rebased](https://git-scm.com/docs/git-rebase) on latest master `git pull --rebase upstream master`
- [x]  All FreeCAD unit tests are confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [x]  All commit messages are [well-written](https://chris.beams.io/posts/git-commit/) ex: `Fixes typo in Draft Move command text`
- [ ]  Your pull request is well written and has a good description, and its title starts with the module name, ex: `Draft: Fixed typos`
- [ ]  Commit messages include `issue #<id>` or `fixes #<id>` where `<id>` is the issue ID number from our [Issues database](https://github.com/FreeCAD/FreeCAD/issues) in case a particular commit solves or is related to an existing issue. Ex: `Draft: fix typos - fixes #4805`

And please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [0.20 Changelog Forum Thread](https://forum.freecadweb.org/viewtopic.php?f=10&t=56135).

---

### CODE REVIEW

**Comment by berndhahnebach** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7168">pipeline status</a> for feature branch [PR_7168](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7168). Pipeline [585323603](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/585323603) was triggered at [8f95985](https://github.com/FreeCAD/FreeCAD/pull/7168/commits/8f95985a9a3aa8d368ddfa78f0803c9bd8b90498). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

File: src/Mod/Path/PathScripts/PathToolBitLibraryGui.py [hunk 1]
```diff
@@ -878,3 +894,71 @@ def libararySaveLinuxCNC(self, path):
...
+
+                if hasattr(bit, "Camotics"):
+                    toolitem["shape"] = bit.Camotics
+                else:
+                    toolitem["shape"] = SHAPEMAP.get(bit.ShapeName, "Cylindrical")
+
+                toolitem["units"] = unitstring
+                FreeCAD.ActiveDocument.removeObject(bit.Name)
+
+                toollist[toolNr] = toolitem
```
Comment (benj5378) [inline src/Mod/Path/PathScripts/PathToolBitLibraryGui.py:960]:
```suggestion
            if not bit:
                continue
            
            PathLog.track(bit)

            toolitem = tooltemplate.copy()

            toolitem["diameter"] = (
                float(bit.Diameter.getUserPreferred()[0].split()[0])
                if hasattr(bit, "Diameter")
                else 2
            )
            toolitem["description"] = bit.Label
            toolitem["length"] = (
                float(bit.Length.getUserPreferred()[0].split()[0])
                if hasattr(bit, "Length")
                else 10
            )

            if hasattr(bit, "Camotics"):
                toolitem["shape"] = bit.Camotics
            else:
                toolitem["shape"] = SHAPEMAP.get(bit.ShapeName, "Cylindrical")

            toolitem["units"] = unitstring
            FreeCAD.ActiveDocument.removeObject(bit.Name)

            toollist[toolNr] = toolitem
```

**Comment by berndhahnebach** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7168">pipeline status</a> for feature branch [PR_7168](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7168). Pipeline [586112380](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/586112380) was triggered at [792cb05](https://github.com/FreeCAD/FreeCAD/pull/7168/commits/792cb05473d459bfefac666be9e84351c59f48b7). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by berndhahnebach** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7168">pipeline status</a> for feature branch [PR_7168](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7168). Pipeline [589539499](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/589539499) was triggered at [fa49c07](https://github.com/FreeCAD/FreeCAD/pull/7168/commits/fa49c07840fd86d91c225fb3272f2ab987f6e84d). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Pull request merged**